### PR TITLE
Drop babel transpiling and `core-js` dependency

### PR DESCRIPTION
**Pull request by susnux**
Simply use typescript to also emit javascript for common-js and check emitted javascript to be compatible with our `browserslist` targets by using `check-es-compat`.

---

This is a follow up on the discussion of #535, but lets wait on https://github.com/nextcloud/standards/issues/10

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Jest doesn't approve! :see_no_evil:

**Comment by skjnldsv** [comment]
https://jestjs.io/docs/getting-started#via-ts-jest

**Comment by skjnldsv** [comment]
Done, please review my last commit @susnux

File: package.json [hunk 1]
```diff
@@ -34,30 +35,22 @@
...
     "escape-html": "^1.0.3",
     "node-gettext": "^3.0.0"
   },
   "devDependencies": {
-    "@babel/cli": "^7.19.3",
-    "@babel/core": "^7.20.5",
-    "@babel/preset-env": "^7.20.2",
-    "@babel/preset-typescript": "^7.18.6",
-    "@happy-dom/jest-environment": "^8.1.0",
     "@nextcloud/browserslist-config": "^2.3.0",
```
Comment (skjnldsv) [inline package.json:43]: can be removed too

Comment (susnux) [reply package.json:43]: Browserslist config? This is still used by `check-es-compat`

File: package.json [hunk 2]
```diff
@@ -35,26 +35,23 @@
...
   "devDependencies": {
-    "@happy-dom/jest-environment": "^8.1.0",
     "@nextcloud/browserslist-config": "^2.3.0",
     "@nextcloud/typings": "^1.0.0",
     "@types/node-gettext": "^3.0",
     "check-es-compat": "^2.2.0",
     "gettext-parser": "^4.0.3",
-    "jest": "^27.2.1",
+    "jest": "^29.3.1",
+    "jest-environment-jsdom": "^29.3.1",
```
Comment (susnux) [inline package.json:49]: What is the reason for switching to jsdom?
(Asking because happydom should be much faster)

Comment (skjnldsv) [reply package.json:49]: Seemed to be recommended with ts-jest and inspired myself from the other configs that works :)
But I have absolutely zero preferences, feel free to switch to happydom. As far as I understood jsdom seemed a bit more mature, but I have little expertise on the subject.

The rest of my config  change should still be relevant :wink:

File: package.json (hunk 1 above)
Comment (skjnldsv) [reply package.json:43]: :+1:

File: package.json (hunk 2 above)
Comment (susnux) [reply package.json:49]: No I am fine with it, I was just curious if there is a reason I do not know :)

**Comment by susnux** [comment]
> Done, please review my last commit @susnux

@skjnldsv: Looks good, I just added the compatibility check to the CI pipeline.
See https://github.com/nextcloud/nextcloud-l10n/pull/544/commits/d3986684334b61ead93b388128e0355f97898fd4

**Pull request merged**